packages/components/hooks/use-scroll-position.ts

```diff
@@ -0,0 +1,29 @@
+import { useRef, useState } from 'react'
+
+import { useIsomorphicLayoutEffect } from './use-isomorphic-layout-effect'
+
+interface ScrollPosition {
+  ref: React.RefObject<HTMLDivElement>
+  scrollPos: number
+}
+
+const useScrollPosition = (): ScrollPosition => {
+  const ref = useRef<HTMLDivElement>(null)
+  const [scrollPos, setScrollPos] = useState(0)
+
+  useIsomorphicLayoutEffect(() => {
+    const newRef = ref.current
```

Comment:
I don't think `newRef` is a proper name because nothing is new. I suggest using `element` name because that's what the ref is referencing.